tests/sentry/api/endpoints/test_sentry_app_details.py

```diff
@@ -303,6 +304,14 @@ def test_no_webhook_has_alerts(self):
             "webhookUrl": ["webhookUrl required if alert rule action is enabled"]
         }
 
+    def test_set_allowed_origins(self):
+        self.login_as(user=self.user)
+        response = self.client.put(
+            self.url, data={"allowedOrigins": ["google.com", "sentry.io"]}, format="json"
```

Comment:
Should these values be validated to ensure they look like hostnames? Right now I don't see anything that would prevent someone providing `*` as a value or an invalid hostname?

Reply:
@markstory We have no such validation for ApiApplications. My opinion is no. If you want your app to have an allowed origin of `*`, that's your call as the app is yours. And if you provide an invalid hostname, well then that value isn't going to do anything. But I can't imagine it would break anything.

Reply:
I disagree. I think allowing people to use `*` is just setting ourselves up for headaches and making our system too vulnerable. Even if _they're_ okay with the elevated risk of exposing their own data this way, _we're_ ultimately responsible for the result, if that happens.

I'd rather require people to specify some sort of specific URL. Maybe it can _include_ `*`, eg. `*.mycoolthing.com`, but `*` by itself seems too permissive to me. Even if ApiApplications allow it.

@scefali @markstory Thoughts?

Reply:
@mnoble Ok, that makes sense. My concern is making the validation logic too complicated. Let's try to come up with the rules we want to enforce here together.